src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -341,11 +362,66 @@ def run_single(
                 status=SCMStatus.ANALYSIS_FAILED,
             )
 
+        # Prepare the repo.
+        git_obj = None
+        if analysis_target.repo_path:
+            git_obj = self._prepare_repo(
+                os.path.join(self.output_path, self.GIT_REPOS_DIR),
+                analysis_target.repo_path,
+                analysis_target.branch,
+                analysis_target.digest,
+                analysis_target.parsed_purl,
+            )
+
+        # Check if only one of the repo or digest came from direct input, and a versioned purl exists for the other.
+        if (provenance_repo_url or provenance_commit_digest) and parsed_purl and parsed_purl.version:
+            if determine_abstract_purl_type(parsed_purl) == AbstractPurlType.REPOSITORY:
+                if not repo_path_input and provenance_repo_url:
+                    # Check the repo from purl.
+                    possibly_expanded_type = parsed_purl.type
+                    if defaults.has_section("package_url"):
```

Comment:
I wonder if resolving the mapping between the PURL type for git services (e.g. `github` or `bitbucket`) to its domain name is in the same manner as handled by this function https://github.com/oracle/macaron/blob/f21bb6fb38a10fee47b0512f07d7ecfb3fb05da2/src/macaron/repo_finder/repo_finder.py#L81 Or are these mappings in `defaults.ini` also serve another purpose :thinking:

Reply:
There is sort of an argument for having it be user configurable, but that can be a discussion for another time. I will use this instead.

Reply:
Oh okay, so there is a reason behind this decision. I will leave it to you for the final decision. Both ways are totally good for me. Thanks for the clarification.